### PR TITLE
SDIT-2110 Handle merges for casenotes - booking

### PR DESCRIPTION
**Pull request by steverendell**
booking move merge

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/casenotes/CaseNoteMappingRepository.kt [hunk 1]
```diff
@@ -34,4 +34,7 @@ interface CaseNoteMappingRepository : CoroutineCrudRepository<CaseNoteMapping, S
   @Modifying
   @Query("UPDATE CASE_NOTE_MAPPING SET offender_no = :toOffenderNo WHERE offender_no = :fromOffenderNo")
   suspend fun updateOffenderNo(fromOffenderNo: String, toOffenderNo: String): Int
+
+  @Query("UPDATE CASE_NOTE_MAPPING SET offender_no = :toOffenderNo WHERE nomis_booking_id = :bookingId returning *")
+  suspend fun updateOffenderNoByBooking(bookingId: Long, toOffenderNo: String): List<CaseNoteMapping>
```
Comment (steverendell) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/casenotes/CaseNoteMappingRepository.kt:39]: Im amazed this 'returning' malarkey just worked 🤷

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/casenotes/CaseNoteMappingService.kt [hunk 2]
```diff
@@ -175,7 +175,7 @@ class CaseNoteMappingService(
   suspend fun updateMappingsByNomisId(oldOffenderNo: String, newOffenderNo: String) {
     val count = repository.updateOffenderNo(oldOffenderNo, newOffenderNo)
     telemetryClient.trackEvent(
-      "nonAssociation-mapping-merged",
+      "casenotes-mapping-prisoner-merged",
```
Comment (steverendell) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/casenotes/CaseNoteMappingService.kt:178]: from previous pr

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/casenotes/CaseNoteMappingResourceIntTest.kt [hunk 3]
```diff
@@ -159,7 +158,6 @@ class CaseNoteMappingResourceIntTest : IntegrationTestBase() {
           .contentType(MediaType.APPLICATION_JSON)
           .body(
             BodyInserters.fromValue(
-              //language=JSON
```
Comment (steverendell) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/casenotes/CaseNoteMappingResourceIntTest.kt:162]: removed these as they were confusing intellij

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**